Userland/Libraries/LibWasm/Parser/LEB128.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Stream.h>
+#include <AK/Types.h>
+
+namespace Wasm {
+
+struct LEB128 {
```

Comment:
There's LEB128 code in LibDebug already, can we find a way to share this?

Reply:
I assume you mean the one in `AK/MemoryStream.h`?
This is actually a copy of that, but that one wants to un-read whatever it has read when it fails.
I neither need nor want that behaviour here.

We could implement it in `InputStream`, but that seems like a really odd thing to add to a stream.
One thing I could do is to move it to `AK/LEB128.h` and make both impls use that

Reply:
>One thing I could do is to move it to `AK/LEB128.h` and make both impls use that

👍🏻